precompiles/bech32/methods_test.go

```diff
@@ -0,0 +1,202 @@
+package bech32_test
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/evmos/v15/cmd/config"
+	"github.com/evmos/evmos/v15/precompiles/bech32"
+	cmn "github.com/evmos/evmos/v15/precompiles/common"
+)
+
+func (s *PrecompileTestSuite) TestHexToBech32() {
+	// setup basic test suite
+	s.SetupTest()
+
+	method := s.precompile.Methods[bech32.HexToBech32Method]
+
+	testCases := []struct {
+		name        string
+		malleate    func() []interface{}
+		postCheck   func(data []byte)
+		expError    bool
+		errContains string
+	}{
+		{
+			"fail - invalid args length",
+			func() []interface{} {
+				return []interface{}{}
+			},
+			func(data []byte) {},
+			true,
+			fmt.Sprintf(cmn.ErrInvalidNumberOfArgs, 2, 0),
+		},
+		{
+			"fail - invalid hex address",
+			func() []interface{} {
+				return []interface{}{
+					"",
+					"",
+				}
+			},
+			func(data []byte) {},
+			true,
+			"invalid hex address",
+		},
+		{
+			"fail - invalid bech32 HRP",
+			func() []interface{} {
+				return []interface{}{
+					s.keyring.GetAddr(0),
+					"",
+				}
+			},
+			func(data []byte) {},
+			true,
+			"invalid bech32 human readable prefix (HRP)",
+		},
+		{
+			"pass - valid hex address and valid bech32 HRP",
+			func() []interface{} {
+				return []interface{}{
+					s.keyring.GetAddr(0),
+					config.Bech32Prefix,
+				}
+			},
+			func(data []byte) {
+				args, err := s.precompile.Unpack(bech32.HexToBech32Method, data)
+				s.Require().NoError(err, "failed to unpack output")
+				s.Require().Len(args, 1)
+				addr, ok := args[0].(string)
+				s.Require().True(ok)
+				s.Require().Equal(s.keyring.GetAccAddr(0).String(), addr)
+			},
+			false,
+			"",
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			s.SetupTest()
+
+			bz, err := s.precompile.HexToBech32(&method, tc.malleate())
+
+			if tc.expError {
+				s.Require().Error(err)
+				s.Require().ErrorContains(err, tc.errContains, err.Error())
+				s.Require().Empty(bz)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NotEmpty(bz)
+				tc.postCheck(bz)
+			}
+		})
+	}
+}
+
+func (s *PrecompileTestSuite) TestBech32ToHex() {
+	// setup basic test suite
+	s.SetupTest()
+
+	method := s.precompile.Methods[bech32.Bech32ToHexMethod]
+
+	testCases := []struct {
+		name        string
+		malleate    func() []interface{}
+		postCheck   func(data []byte)
+		expError    bool
+		errContains string
+	}{
+		{
+			"fail - invalid args length",
+			func() []interface{} {
+				return []interface{}{}
+			},
+			func(data []byte) {},
+			true,
+			fmt.Sprintf(cmn.ErrInvalidNumberOfArgs, 1, 0),
+		},
+		{
+			"fail - empty bech32 address",
```

Comment:
nit: in the analogous test for HexToBech it is called "invalid hex address". For consistency this one could be "invalid bech32 address (or change the other)